File: ocaml/numbers/test/test_numbers_iterative.ml
open OUnit2
module NumbersIterative = Numbers.Numbers_iterative

let test_recursive = "Iterative Tests" >:::
  [
    "test_fibonacci" >:: (fun _ ->
      assert_equal 5 (NumbersIterative.fibonacci 5);
    );

    "test_factorial" >:: (fun _ ->
      assert_equal 120 (NumbersIterative.factorial 5);
    );

    "test_sum_numbers" >:: (fun _ ->
      assert_equal 15 (NumbersIterative.sum_numbers 5);
    );
  ]

let () =
  run_test_tt_main test_recursive

File: _experimental/_nim/numbers/tests/test_numbers_iterative.nim
import unittest
import numbers_iterative

suite "Iterative Tests":
    test "fibonacci of 5":
        assert fibonacci(5) == 5
    
    test "factorial of 5":
        assert factorial(5) == 120

    test "sum_numbers of 5":
        assert sum_numbers(5) == 15